### PR TITLE
Regex - Optional Unicode properties support, and prefer brewed libpcre on MacOS

### LINKED ISSUE #4704: Regex can't match utf-8 characters
Hello,

I was trying to match some utf-8 characters but it doesn't seem to work.

```crystal
/[[:alnum:]]/x.match "à" # => nil
```

It works in Ruby and seems compliant with PCRE, is there a bug with crystal or is there a subtlety I didn't notice here ?

### PR DESCRIPTION
**Pull request by akzhan**
Current crystal depends on `pcre` instead of `libpcre`. It breaks linking to newest PCRE under MacOS. 

I have changed link attribute and add specs for UCP support.

```crystal
/(*UCP)[[:alnum:]]/.match "à"
```

UCP support disabled by default due to bad performance, but can be used by "(*UCP)" flag or by `Regex::Options::UCP` option.

This pull request depends on PCRE 8.10 or above.

Fixes #4704.

### CODE REVIEW

**Comment by akzhan** [comment]
CrcleCI has too old PCRE.

**Comment by RX14** [comment]
We should find a way to degrade gracefully as we do with openssl. Use `pkgconfig` to query the library version etc.

**Comment by akzhan** [comment]
@RX14 thanks, I'll do. Anyway it should be optional flag.

**Comment by akzhan** [comment]
CircleCI is strange again. std spec built ok, but all_spec not.

File: spec/std/regex_spec.cr [hunk 1]
```diff
@@ -82,9 +82,18 @@ describe "Regex" do
...
+  if LibPCRE::HAS_UCP
+    it "matches with Unicode properties for alnum character class with UCP flag" do
+      (/(*UCP)[[:alnum:]]/.match "à").should be_truthy
+    end
+
+    it "matches with Unicode properties for \w with UCP option" do
+      (Regex.new("\\w", Regex::Options::UCP).match "à").should be_truthy
+    end
+  else
+    pending "PCRE built without Unicode properties support" do
```
Comment (RX14) [inline spec/std/regex_spec.cr]: I don't think this is necessary. Is it done anywhere else in specs?

Comment (akzhan) [inline spec/std/regex_spec.cr]: Nowhere. It's just declaration that sometimes Crystal will be built with UCP-friendly PCRE only.

**Comment by akzhan** [comment]
I'll close this pull request because want to change regex engine to PCRE2 successor that built with UCP support by default.

**Pull request closed without merging**